src/Base/UnitPyImp.cpp

```diff
@@ -34,23 +34,24 @@ using namespace Base;
 // returns a string which represents the object e.g. when printed in python
 std::string UnitPy::representation() const
 {
-    const UnitSignature& Sig = getUnitPtr()->getSignature();
     std::stringstream ret;
+    Unit* self = getUnitPtr();
+
     ret << "Unit: ";
-    ret << getUnitPtr()->getString().toUtf8().constData() << " (";
-    ret << Sig.Length << ",";
-    ret << Sig.Mass << ",";
-    ret << Sig.Time << ",";
-    ret << Sig.ElectricCurrent << ",";
-    ret << Sig.ThermodynamicTemperature << ",";
-    ret << Sig.AmountOfSubstance << ",";
-    ret << Sig.LuminousIntensity << ",";
-    ret << Sig.Angle << ")";
-    std::string type = getUnitPtr()->getTypeString().toUtf8().constData();
+    ret << self->getString() << " (";
+    ret << (*self)[0] << ",";
```

Comment:
With the old code the meaning of the bit fields were quite clear because they were member variables of the UnitSignature struct. But with the new code the meaning of the index value isn't clear any more. One has to always check the implementation to see what the numbers 0-7 relate to.

And from the API it's not even obvious that the value range is [0, 7].

What do you think about adding the member functions:
```
int length() const
int mass() const
int time() const
...
```
?

Reply:
I agree this is sloppy API. I'm not fan of exposing too much internals until necessary, so I'll try to make UnitPy a descendant of Unit (it is generated, I know nothing about it yet, but worth trying).

Reply:
Alternatively UnitPy could be descendant of Unit class, but it is generated and I do not know it is even possible. Base units are also used in other places of code, so you are probably right.